osu.Game/Screens/SelectV2/BeatmapCarousel.cs

```diff
@@ -360,13 +360,22 @@ protected override float GetPanelXOffset(Drawable panel)
 
         private ScheduledDelegate? loadingDebounce;
 
+        /// <summary>
+        /// We need to track this state locally since `FilterCriteria` is passed by reference and not accurate.
+        /// It should really be a struct.
+        /// </summary>
+        private bool splitOutDifficulties;
```

Comment:
I don't follow. What does `FilterCriteria` being passed by reference or by value have to do with the "accuracy" of state?

Reply:
In the new usage, I compare the state of the `bool`. If I don't store a local, it's always equal, since everything is operating on the *same instance of `FilterCriteria`*.

Yes, this is super dangerous and wrong and help.

Reply:
Correct me if I'm wrong, but the filter control always generates a new filter criteria instance whenever any control is updated. Is this not the case somehow?

Reply:
good question - not in tests it doesn't!

https://github.com/ppy/osu/blob/790a2f8f37abfc88e5b51faa15d7db20875ceaf8/osu.Game.Tests/Visual/SongSelectV2/BeatmapCarouselTestScene.cs#L160-L168